src/ui/buttons/buttonDesigns/alternateSlideLogoDesign.jsx

```diff
@@ -0,0 +1,209 @@
+    const animationContainerWidth = designContainer.offsetWidth;
+    if (animationContainerWidth < min || animationContainerWidth > max) {
+        // remove label element from dom
+        designContainer.querySelector(`.${ ANIMATION_LABEL_CONTAINER }`).remove();
+        return null;
+    }
+
+    // get the label container that animation will be applied to
+    const paypalLabelContainerElement = designContainer.querySelector(`.${ PAYPAL_BUTTON_LABEL }`) || null;
+    // get starting position for element so it doesn't jump when animation begins
+    const paypalLogoElement = (paypalLabelContainerElement && paypalLabelContainerElement.querySelector(`.${ PAYPAL_LOGO }`)) || null;
+    
+    if (paypalLogoElement && paypalLabelContainerElement) {
+        paypalLogoStartingLeftPosition = `${ (paypalLogoElement.offsetLeft / paypalLabelContainerElement.offsetWidth) * 100 }`;
+    } else {
+        return null;
+    }
+    // increase font-size for medium button sizes
+    const labelTextFontSize = animationContainerWidth >= smalMax ? 11  : 8;
+
+    if (runOnce) {
+        // enable reverse logo animation on mouser out
+        designContainer.addEventListener('mouseleave', () => {
+            paypalLogoElement.style.animationName = 'reverse-logo';
+            paypalLogoElement.style.animationDuration = '1s';
+            paypalLogoElement.style.animationPlayState = 'running';
+        }, false);
+    
+        // enable animation from center to left on mouse enter
+        designContainer.addEventListener('mouseenter', () => {
+            paypalLogoElement.style.animationName = 'move-logo-to-left-side';
+            paypalLogoElement.style.animationPlayState = 'running';
+        }, false);
+    }
+   
+    return {
+        designContainer,
+        paypalLabelContainerElement,
+        paypalLogoStartingLeftPosition,
+        labelTextFontSize
+    };
+};
+
+export function getAlternateSlideLogoAnimation (designProps : AnimationProps, configuration : Object) : void | null {
+    const { ANIMATION_LABEL_CONTAINER, ANIMATION_CONTAINER, DOM_READY, PAYPAL_LOGO } = configuration.cssClasses;
+    const { max, min, runOnce } = configuration;
+    const { designContainer, paypalLabelContainerElement, paypalLogoStartingLeftPosition, labelTextFontSize } = designProps;
+
+    const playOnHover = runOnce ? ':hover' : '';
+    const animationPropertiesOnHover = `
+        animation-duration: 2s;
+        animation-fill-mode: forwards;
+    `;
+    const keyFramesOnHover = `
+        0%,33%{
+            opacity: 0;
+        }
+    `;
+    const keyframesWithoutHover = `
+        0%,33%,66%{
+            opacity: 0;
+        }
+    `;
+    const animationPropertiesWithoutHover = `
+        animation-duration: 3s;
+        animation-iteration-count: infinite;
+        animation-direction: alternate;
+    `;
+    
+    const animationProperties =  playOnHover ? animationPropertiesOnHover : animationPropertiesWithoutHover;
```

Comment:
Can you use the runOnce flag instead here?

Reply:
sure, no problem